### PR TITLE
feat: inputNumber controls support customize upIcon/downIcon

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/33641
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   feat: inputNumber support customize upIcon/downIcon        |
| 🇨🇳 Chinese |  功能:  `inputNumber `组件支持自定义 `向上icon`  `向下icon`       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by heiyu4585** [comment]
抱歉,失误较多

File: components/input-number/__tests__/index.test.js [hunk 1]
```diff
@@ -28,4 +29,37 @@ describe('InputNumber', () => {
     expect(onStep).toBeCalledTimes(2);
     expect(onStep).toHaveBeenLastCalledWith(1, { offset: 1, type: 'down' });
   });
+
+  it('renders correctly when controls is boolean', () => {
+    expect(mount(<InputNumber controls={false}/>).render()).toMatchSnapshot();
+  });
+
+  it('renders correctly when controls is {}', () => {
+    expect(mount(<InputNumber controls={{}}/>).render()).toMatchSnapshot();
```
Comment (afc163) [inline components/input-number/__tests__/index.test.js:38]:
```suggestion
    expect(mount(<InputNumber controls={{}} />).render()).toMatchSnapshot();
```

File: components/input-number/__tests__/index.test.js [hunk 2]
```diff
@@ -28,4 +29,37 @@ describe('InputNumber', () => {
...
+      />,
+    );
+    expect(wrapper.render()).toMatchSnapshot();
+  });
+
+  it('should support className', () => {
+    const wrapper = mount(
+      <InputNumber
+        controls={{
+          upIcon: <ArrowUpOutlined className="my-class-name"/>,
```
Comment (afc163) [inline components/input-number/__tests__/index.test.js:57]:
```suggestion
          upIcon: <ArrowUpOutlined className="my-class-name" />,
```

File: components/input-number/demo/controls.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+When you need to use a custom `Icon`, you can set the `Icon` component as the property value of `upIcon` and `downIcon`
+
+```jsx
+import { InputNumber } from 'antd';
+import { ArrowUpOutlined, ArrowDownOutlined } from '@ant-design/icons';
+
+ReactDOM.render(
+  <InputNumber
+    controls={{ upIcon: <ArrowUpOutlined />, downIcon: <ArrowDownOutlined/>} }
```
Comment (afc163) [inline components/input-number/demo/controls.md:22]:
```suggestion
    controls={{ upIcon: <ArrowUpOutlined />, downIcon: <ArrowDownOutlined/> }}
```

File: components/input-number/demo/controls.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  />
+  , mountNode);
```
Comment (afc163) [inline components/input-number/demo/controls.md:24]:
```suggestion
, mountNode);
```

File: components/input-number/index.tsx [hunk 5]
```diff
@@ -37,12 +38,22 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
     readOnly,
+    controls,
     ...others
   } = props;
 
   const prefixCls = getPrefixCls('input-number', customizePrefixCls);
-  const upIcon = <UpOutlined className={`${prefixCls}-handler-up-inner`} />;
-  const downIcon = <DownOutlined className={`${prefixCls}-handler-down-inner`} />;
+  let upIcon = <UpOutlined className={`${prefixCls}-handler-up-inner`} />
+  let downIcon = <DownOutlined className={`${prefixCls}-handler-down-inner`} />
```
Comment (afc163) [inline components/input-number/index.tsx:47]:
```suggestion
  let upIcon = <UpOutlined className={`${prefixCls}-handler-up-inner`} />;
  let downIcon = <DownOutlined className={`${prefixCls}-handler-down-inner`} />;
```

File: components/input-number/index.tsx [hunk 6]
```diff
@@ -37,12 +38,22 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
+
+  if (typeof controls === 'object') {
+    upIcon = typeof controls.upIcon === 'undefined' ? upIcon : (
+      <span className={`${prefixCls}-handler-up-inner`}>{controls.upIcon}</span>
+    ) ;
```
Comment (afc163) [inline components/input-number/index.tsx:52]:
```suggestion
    upIcon = (
      <span className={`${prefixCls}-handler-up-inner`}>{typeof controls.upIcon === 'undefined' ? upIcon : controls.upIcon}</span>
    ) ;
```

File: components/input-number/index.tsx [hunk 7]
```diff
@@ -64,6 +75,7 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
       downHandler={downIcon}
       prefixCls={prefixCls}
       readOnly={readOnly}
+      controls={typeof controls === 'boolean' ? controls : undefined}
```
Comment (afc163) [inline components/input-number/index.tsx:78]:
```suggestion
      controls={!!controls}
```

**Comment by afc163** [comment]
不需要重新提交，原分支修改后 push 即可。

File: components/input-number/index.tsx (hunk 7 above)
Comment (heiyu4585) [reply components/input-number/index.tsx]:
```
  let upIcon = <UpOutlined className={`${prefixCls}-handler-up-inner`} />;
  let downIcon = <DownOutlined className={`${prefixCls}-handler-down-inner`} />;

  if (typeof controls === 'object') {
    upIcon = (
      <span className={`${prefixCls}-handler-up-inner`}>
        {typeof controls.upIcon === 'undefined' ? <UpOutlined /> : controls.upIcon}
      </span>
    ) ;
 .....
```
此处未改这样改会出现两个`${prefixCls}-handler-up-inner`类,

File: components/input-number/index.tsx (hunk 6 above)
Comment (heiyu4585) [reply components/input-number/index.tsx:52]: > 不需要重新提交，原分支修改后 push 即可。

已push

**Comment by heiyu4585** [comment]
> 不需要重新提交，原分支修改后 push 即可。

已push

File: components/input-number/__tests__/index.test.js [hunk 8]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { mount } from 'enzyme';
+import { ArrowUpOutlined,ArrowDownOutlined } from '@ant-design/icons';
```
Comment (afc163) [inline components/input-number/__tests__/index.test.js:3]:
```suggestion
import { ArrowUpOutlined, ArrowDownOutlined } from '@ant-design/icons';
```

提交时不要 -n，走一下 prettier。

File: components/input-number/index.tsx [hunk 9]
```diff
@@ -37,12 +38,22 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
   const prefixCls = getPrefixCls('input-number', customizePrefixCls);
-  const upIcon = <UpOutlined className={`${prefixCls}-handler-up-inner`} />;
-  const downIcon = <DownOutlined className={`${prefixCls}-handler-down-inner`} />;
+  let upIcon = <UpOutlined className={`${prefixCls}-handler-up-inner`} />;
+  let downIcon = <DownOutlined className={`${prefixCls}-handler-down-inner`} />;
+
+  if (typeof controls === 'object') {
+    upIcon = typeof controls.upIcon === 'undefined' ? upIcon : (
+      <span className={`${prefixCls}-handler-up-inner`}>{controls.upIcon}</span>
+    ) ;
```
Comment (afc163) [inline components/input-number/index.tsx:52]:
```suggestion
    );
```

File: components/input-number/demo/controls.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  en-US: Icon
+---
+
+## zh-CN
+
+可以扩展 `controls` 属性用以设置自定义图标。
+
+## en-US
+
+When you need to use a custom `Icon`, you can set the `Icon` component as the property value of `upIcon` and `downIcon`
```
Comment (afc163) [inline components/input-number/demo/controls.md:14]:
```suggestion
When you need to use a custom `Icon`, you can set the `Icon` component as the property value of `upIcon` and `downIcon`.
```

**Comment by afc163** [comment]
snapshot 需要更新一下。

File: components/input-number/demo/controls.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
+---
+order: 1
+title:
+  zh-CN: 图标按钮
+  en-US: Icon
```
[the remark this answers is not in the document]
Comment (afc163) [reply components/input-number/demo/controls.md:5]: order 也改 99 吧，放到后面去。

**Comment by heiyu4585** [comment]
> snapshot 需要更新一下。

1. 已加debug
2. order已改
3. snapshot已更新
4. prettier已开启

File: components/input-number/index.tsx [hunk 12]
```diff
@@ -37,12 +38,27 @@ const InputNumber = React.forwardRef<HTMLInputElement, InputNumberProps>((props,
...
+      <span className={`${prefixCls}-handler-up-inner`}>
+        {typeof controls.upIcon === 'undefined' ? <UpOutlined /> : controls.upIcon}
+      </span>
+    );
+    downIcon = (
+      <span className={`${prefixCls}-handler-down-inner`}>
+        {typeof controls.upIcon === 'undefined' ? <DownOutlined /> : controls.downIcon}
+      </span>
+    );
+  }
```
Comment (afc163) [inline components/input-number/index.tsx:61]: 好像有点问题，controls 为 `{{}}` 和 controls 为 true 应该是一样的。

Comment (afc163) [reply components/input-number/index.tsx:61]: 之前的写法没问题。

**Review by afc163 — APPROVED** [review]

**Pull request merged**